app/web/features/badges/BadgesPage.tsx

```diff
@@ -0,0 +1,145 @@
+import {
+  Divider,
+  DividerProps,
+  styled,
+  Typography,
+  useMediaQuery,
+  useTheme,
+} from "@mui/material";
+import Button from "components/Button";
+import CenteredSpinner from "components/CenteredSpinner/CenteredSpinner";
+import HtmlMeta from "components/HtmlMeta";
+import PageTitle from "components/PageTitle";
+import UsersList from "components/UsersList";
+import Badge from "features/badges/Badge";
+import { useBadges, useBadgeUsers } from "features/badges/hooks";
+import { useTranslation } from "i18n";
+import { GLOBAL, PROFILE } from "i18n/namespaces";
+
+const BadgeList = styled("div")(({ theme }) => ({
+  // display: "inline-block"
+}));
+
+const BadgeListItem = styled("div")(({ theme }) => ({
+  [theme.breakpoints.down("md")]: {
+    display: "inline-block",
+  },
+}));
+
+const StyledDivider = styled(Divider)<DividerProps>(({ theme }) => ({
+  margin: theme.spacing(2),
+}));
+
+const FlexDiv = styled("div")(({ theme }) => ({
+  display: "flex",
+  gap: theme.spacing(2),
+  alignItems: "start",
+}));
+
+const ParentFlexDiv = styled(FlexDiv)(({ theme }) => ({
+  [theme.breakpoints.down("md")]: {
+    flexDirection: "column",
+    gap: theme.spacing(0),
+  },
+}));
+
+const ContentDiv = styled("div")(({ theme }) => ({
+  padding: theme.spacing(2),
+  alignSelf: "stretch",
+  width: "100%",
+}));
+
+const CenteredDiv = styled(ContentDiv)(({ theme }) => ({
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "center",
+}));
+
+export interface BadgeUserListProps {
+  badgeId: string;
+}
+
+function BadgeUserList({ badgeId }: BadgeUserListProps) {
+  const { t } = useTranslation([PROFILE]);
+
+  const { badgeUserIds, hasNextPage, isFetchingNextPage, fetchNextPage } =
+    useBadgeUsers(badgeId);
+
+  return (
+    <UsersList
+      userIds={badgeUserIds}
+      endChildren={
+        hasNextPage && (
+          <Button loading={isFetchingNextPage} onClick={() => fetchNextPage()}>
+            Load more
+          </Button>
+        )
+      }
+      emptyListChildren={
+        <Typography variant="body1">{t("profile:badges.no_people")}</Typography>
+      }
+    />
+  );
+}
+
+export interface BadgesPageProps {
+  badgeId?: string;
+}
+
+export default function BadgesPage({ badgeId = undefined }: BadgesPageProps) {
```

Comment:
Might make sense to make BadgesPage a separate file? Easier to scroll through I think.

Reply:
What do you mean, sorry?

Reply:
Ah instead of having two components in one file, break it into a separate file.